### PR TITLE
Better Heartbeating External Documentation.

### PR DESCRIPTION
**Pull request by davinchia**
In,
1) https://github.com/airbytehq/airbyte-platform-internal/commit/96baf5b990650fb32a332d6d6baa063f673e3699
2) https://github.com/airbytehq/airbyte-platform-internal/pull/11595

we improve our heartbeat error messages and point users to this external document.

Here, we improve external documentation to help users understand what is happening and what they can do.

### CODE REVIEW

**Review by benmoriceau — APPROVED** [review]

File: docs/understanding-airbyte/heartbeats.md [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
+Airbyte aims to make restarts as automated as possible and uses heartbeating mechanism to do so. This is performed on 2 different components: the source and the destination.
 
-### Heartbeating logic
+Heartbeat errors are expected to be transient and should automatically resolve. If they do not, it is likely a sign of a more serious issue.
+
+## Known Causes
+
+Possible reasons for this issue:
+1. Certain API sources take an unknown amount of time to generate an asynchronous responses (e.g., Salesforce, Facebook, Amplitude). No workaround currently exists.
+2. Certain API sources can be rate-limited for a time period longer than their configured threshold. Although Airbyte tries our best to handle this on a per-connector basis, rate limits are not always predictable.
```
Comment (tryangul) [inline docs/understanding-airbyte/heartbeats.md:13]: Airbyte tries our best -> Airbyte tries _its_ best

File: docs/understanding-airbyte/heartbeats.md [hunk 2]
```diff
@@ -1,34 +1,55 @@
...
-## Source
+Airbyte aims to make restarts as automated as possible and uses heartbeating mechanism to do so. This is performed on 2 different components: the source and the destination.
 
-### Heartbeating logic
+Heartbeat errors are expected to be transient and should automatically resolve. If they do not, it is likely a sign of a more serious issue.
+
+## Known Causes
+
+Possible reasons for this issue:
+1. Certain API sources take an unknown amount of time to generate an asynchronous responses (e.g., Salesforce, Facebook, Amplitude). No workaround currently exists.
```
Comment (tryangul) [inline docs/understanding-airbyte/heartbeats.md:12]: generate an asynchronous responses -> generate ~~an~~ asynchronous responses

File: docs/understanding-airbyte/heartbeats.md [hunk 3]
```diff
@@ -1,34 +1,55 @@
 # Heartbeats
 
-During a data synchronization, many things can go wrong and sometimes the fix is just to restart the synchronization.
-Airbyte aims to make this restart as automated as possible and uses heartbeating mechanism in order to do that.
-This performed on 2 differents component: the source and the destination. They have different logics which will be
-explained bellow.
+Many transient things can go wrong when moving data, especially for long jobs. Often the fix is a simple restart.
 
-## Source
+Airbyte aims to make restarts as automated as possible and uses heartbeating mechanism to do so. This is performed on 2 different components: the source and the destination.
```
Comment (tryangul) [inline docs/understanding-airbyte/heartbeats.md:5]: uses heartbeating mechanism -> uses _a_ heartbeating mechanism

**Review by tryangul — APPROVED** [review]
Left a couple small grammar tweaks but LGTM 🚢

**Pull request merged**